### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Chenjunjie1996**
realize preflight check for multi module

### CODE REVIEW

**Comment by zhouyiqi91** [comment]
1. add package version check
2. consider using loops

**Review by zhouyiqi91 — CHANGES_REQUESTED** [review]
1. add package version check
2. consider using loops

File: celescope/tools/preflight.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+        elif key == 'editdistance':
+            if val != ref_dict[key]:
+                print(msg%(key,ver_dict[key],key,ref_dict[key]))
+        elif key in ref_dict:
+            if ver_dict[key] != ref_dict[key]:
+                print("Warning: package %s version is %s, which doesn't match celescope required version: %s" 
+                      %(key,ver_dict[key],ref_dict[key]))
+
+
+def run_prefligh(mapfile,outdir):
```
Comment (zhouyiqi91) [inline celescope/tools/preflight.py]: why not run_prefligh**t** ?

File: celescope/tools/preflight.py [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+        msg = "You're in the right conda environment."
+    msg += "\n" + "         Make sure celescope and required packages have been installed in current environment successfully."
+    print(msg)
+
+
+def get_version():
+    '''
+    get version cmd
+    '''
+    PACKAGE_VERSION_CMDS = [
```
Comment (zhouyiqi91) [inline celescope/tools/preflight.py]: using loops and function

**Review by zhouyiqi91 — CHANGES_REQUESTED** [review]

File: celescope/tools/preflight.py [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+import re
+import pandas as pd
+
+'''
+Functions for preflight checks 
+'''
+
+@utils.add_log
+def check_file(mapfile):
+    print("Checking mapfile format and raw fastq file path...")
```
Comment (zhouyiqi91) [inline celescope/tools/preflight.py]: use celescope.tools.Multi.parse_mapfile

File: celescope/tools/preflight.py (hunk 1 above)
Comment (Chenjunjie1996) [reply celescope/tools/preflight.py]: Running preflight when calling multi module before celescope workflow.
can be tested in /SGRNJ03/randd/cjj/celedev/preflight_check/test
run shell in env: cjjdev

**Pull request closed without merging**